redis/src/main/scala/zio/redis/Output.scala

```diff
@@ -385,6 +385,152 @@ object Output {
+                else if (key.asString == XInfoFields.RadixTreeNodes)
+                  streamInfoFull = streamInfoFull.copy(radixTreeNodes = value.value)
+                else if (key.asString == XInfoFields.RadixTreeKeys)
+                  streamInfoFull = streamInfoFull.copy(radixTreeKeys = value.value)
+              case (key @ RespValue.BulkString(_), value @ RespValue.BulkString(_))
+                  if key.asString == XInfoFields.LastGeneratedId =>
+                streamInfoFull = streamInfoFull.copy(lastGeneratedId = value.asString)
+              case (key @ RespValue.BulkString(_), RespValue.Array(values)) if key.asString == XInfoFields.Entries =>
+                val streamEntryList = values.map(extractStreamEntry)
+                streamInfoFull = streamInfoFull.copy(entries = streamEntryList)
+              case (key @ RespValue.BulkString(_), RespValue.Array(value)) if key.asString == XInfoFields.Groups =>
+                // Get the group list of the stream.
+                streamInfoFull = streamInfoFull.copy(groups = extractXInfoFullGroups(value))
+              case _ =>
+            }
+            pos += 2
+          }
+          streamInfoFull
+        case array @ RespValue.Array(_) =>
+          throw ProtocolError(s"$array doesn't have an even number of elements")
+        case other =>
+          throw ProtocolError(s"$other isn't an array")
+      }
+    }
+  }
+
+  private def extractXInfoFullGroups(groupValue: Chunk[RespValue]): Chunk[StreamInfoWithFull.ConsumerGroups] = {
+    var groupList: Chunk[StreamInfoWithFull.ConsumerGroups] = Chunk.empty
+    groupValue.foreach { group: RespValue =>
+      var readyGroup = StreamInfoWithFull.ConsumerGroups.empty
+      group match {
+        case RespValue.Array(groupElements) if groupElements.length % 2 == 0 =>
+          var groupElementPos = 0
+          while (groupElementPos < groupElements.length) {
+            (groupElements(groupElementPos), groupElements(groupElementPos + 1)) match {
+              // Get the basic information of the current group.
+              case (key @ RespValue.BulkString(_), value @ RespValue.BulkString(_)) =>
+                if (key.asString == XInfoFields.LastDeliveredId)
+                  readyGroup = readyGroup.copy(lastDeliveredId = value.asString)
+                else if (key.asString == XInfoFields.Name) readyGroup = readyGroup.copy(name = value.asString)
+              case (key @ RespValue.BulkString(_), value @ RespValue.Integer(_))
+                  if XInfoFields.PelCount == key.asString =>
+                readyGroup = readyGroup.copy(pelCount = value.value)
+              case (key @ RespValue.BulkString(_), RespValue.Array(value)) =>
+                // Get the consumer list of the current group.
+                if (key.asString == XInfoFields.Consumers) {
+                  readyGroup = readyGroup.copy(consumers = extractXInfoFullConsumers(value))
```

Comment:
I suggest that you create the `extractXInfoFullConsumer` function and use `values.map` here, just like you did for an `extractStreamEntry`.